### PR TITLE
Important changes for stream

### PR DESCRIPTION
**Pull request by Gallimathias**
Implement the first important changes needed for the NoobDev stream. APIs that will no longer be supported in the future have been marked with an obsolete.

Otherwise, only new lines of code were added.

### CODE REVIEW

File: src/VaVare/Statements/DeclarationStatement.cs [hunk 1]
```diff
@@ -97,23 +167,29 @@ public LocalDeclarationStatementSyntax DeclareAndAssign(string name, Type type,
...
         /// <param name="useVarKeyword">Will use "var" keyword if true, otherwise the type.</param>
         /// <returns>The generated local declaration statement.</returns>
-        public LocalDeclarationStatementSyntax DeclareAndAssign(string name, Type type, ArgumentListSyntax arguments, bool useVarKeyword = true)
+        [Obsolete("Use the overload without optional parameter or without type instead")]
+        public LocalDeclarationStatementSyntax DeclareAndAssign(string name, Type type, ArgumentListSyntax arguments, bool useVarKeyword = true) 
+            => DeclareAndAssign(name, TypeGenerator.Create(type), arguments, useVarKeyword);
+
+        public LocalDeclarationStatementSyntax DeclareAndAssign(string name, TypeSyntax type, ArgumentListSyntax arguments, bool useVarKeyword = true)
         {
             if (string.IsNullOrEmpty(name))
```
Comment (susch19) [inline src/VaVare/Statements/DeclarationStatement.cs:176]: Ich weiß, dass kommt nicht von dir, aber hier sollte doch viel eher string.IsNullOrWhitespace hin, weil " " ist auch kein gültiger Name

Comment (susch19) [reply src/VaVare/Statements/DeclarationStatement.cs:176]: Ich habe gesehen, dass das an mehrere Stellen der Fall ist, eventuell muss man also mal übers ganze Projekt gehen und neue Unit Tests hinzufügen, die den Whitespace Fall mit abdecken

**Review by susch19 — APPROVED** [review]

File: src/VaVare/Statements/DeclarationStatement.cs (hunk 1 above)
Comment (Gallimathias) [reply src/VaVare/Statements/DeclarationStatement.cs:176]: Machen wir am besten in ner zweiten iteration. War jetzt ja eh erst mal nur das nötigste für den stream
